python_modules/dagster/dagster/_core/definitions/reconstruct.py

```diff
@@ -760,9 +743,42 @@ def repository_def_from_target_def(
         if repository_load_data is None:
             return target.compute_repository_definition()
         # can use the cached data to more efficiently load data
-        return target.reconstruct_repository_definition(repository_load_data)
-    else:
-        return None
+        else:
+            return target.reconstruct_repository_definition(repository_load_data)
+    return None
+
+
+def repository_def_from_target_def(
+    target: object,
+    load_type: "DefinitionsLoadType",
+    repository_load_data: Optional["RepositoryLoadData"] = None,
+    set_new_context: bool = True,
+) -> Optional["RepositoryDefinition"]:
```

Comment:
So at this point I think we should have two different functions since they are quite different between the reconstruction and initialization use case. The reconstruction case always has `repository_load_data` set and initialization does not.  And I bet `set_new_context` can be eliminated too.

Also worth nothing that `repository_def_from_pointer` only has a single callsite which is reconstruction-only, so that doesn't need to be parameterized.

Also willing to approve this as is, but I'd like a refactor PR stacked on top as I think this optional parameter + boolean is a recipe for trouble.

Reply:
Updated